### DISCUSSION TITLE
How can server known if OPA have had bundle in long-polling request?

### DISCUSSION THREAD

**Discussion by gitsang** [open-policy-agent/community#641] [Community]
If there is no new bundle, the server responds with a `304 Not Modified` response. But how can server known if opa have had bundle?

When opa don't have bundle yet, server should not response 304, but all request look the same.

**Comment by gitsang** [comment]
When OPA redeploy without cache data. Bundle server will always return 304, even if OPA don't have any bundle.

**Comment by anderseknert** [comment]
OPA sends along an Etag for the revision it currently has of the bundle. So the server will respond with a new bundle if:
* There is no etag in the request, or
* The etag sent in the request doesn't match the bundle revision that the server has
